### DISCUSSION TITLE
Difference between setting current through parameters versus through experiment

### DISCUSSION THREAD

**Discussion by SimMLPhys** [pybamm-team/PyBaMM#3672] [Q&A]
Between using Current Function in parameters and defining it through pybamm.experiment, is there a preferred way to set the current if my simulation's input is the current profile? 

I'm also trying to define a current profile which uses both the string-based instructions ([Pybamm Tutorial 5](https://docs.pybamm.org/en/latest/source/examples/notebooks/getting_started/tutorial-5-run-experiments.html)) and appends a custom current function (like at the end of [Tutorial 4]((https://docs.pybamm.org/en/latest/source/examples/notebooks/getting_started/tutorial-4-setting-parameter-values.html))) at the end of the string-based instructions. Is that possible or am I mixing two different things?

**Comment by valentinsulzer** [comment]
Using the experiment is slightly better as you get access to summary variables. You can mix a custom current function and string based instructions

**Reply by SimMLPhys** [reply]
I'm having trouble figuring out how to combine the custom current function with the string based instructions. For instance, this code and some other variations of it I tried keep giving me one or the other kind of a compatibility error:


```
def create_custom_experiment():
    def my_fun(A, constant):
        def current(t):
            sine_terms = A * pybamm.sin(2 * np.pi * t) 
            return sine_terms + constant
        return current

    experiment_starting = pybamm.Experiment(
        [
            ("Discharge at C/10 for 3 hours or until 3.3 V", "Rest for 1 hour"),
        ] * 3
    )
    experiment_ending = my_fun(5, 0.1)
    return experiment_starting + pybamm.Experiment(experiment_ending)

```
